components/VotePoll.tsx

```diff
@@ -0,0 +1,257 @@
+
+type Props = {
+  pollData: Poll;
+  setPageAlert: Dispatch<SetStateAction<AlertShape | null>>;
+};
+
+function VotePoll({ pollData, setPageAlert }: Props) {
+  const [_, createVote] = useFetch(CreateVote);
+  const [candidates, setCandidates] = useState<string[]>([]);
+  const [chosenCandidates, setChosenCandidates] = useState<string[]>([]);
+
+  useEffect(() => {
+    if (pollData.candidateList) {
+      setCandidates(pollData.candidateList.map((candidate) => candidate.name));
+    }
+  }, [pollData.candidateList]);
+
+  type DropId = "candidatesDropId" | "chosenCandidatesDropId";
+
+  const dropIdMap = {
+    candidatesDropId: {
+      list: candidates,
+      setList: setCandidates,
+    },
+    chosenCandidatesDropId: {
+      list: chosenCandidates,
+      setList: setChosenCandidates,
+    },
+  };
+
+  const onDragEnd = (result: DropResult) => {
+    const { source, destination } = result;
+    if (!destination) return;
+    const sourceId = source.droppableId as DropId;
+    const destId = destination.droppableId as DropId;
+
+    if (sourceId === destId) {
+      const reOrderedItems = reorder(
+        dropIdMap[sourceId].list,
+        source.index,
+        destination.index
+      );
+
+      dropIdMap[sourceId].setList(reOrderedItems);
+    } else {
+      // Item is being moved from one list to another
+      if (!isValidMove(destId)) {
+        // Create a Toast
+        return;
+      }
+      const [newSourceList, newDestList] = move(
+        dropIdMap[sourceId].list,
+        dropIdMap[destId].list,
+        source.index,
+        destination.index
+      );
+
+      dropIdMap[sourceId].setList(newSourceList);
+      dropIdMap[destId].setList(newDestList);
+    }
+  };
+
+  const reorder = (
+    list: string[],
+    startIndex: number,
+    endIndex: number
+  ): string[] => {
+    const result = Array.from(list);
+    const [removed] = result.splice(startIndex, 1);
+    result.splice(endIndex, 0, removed);
+    return result;
+  };
+
+  const isValidMove = (destId: DropId): boolean => {
+    const destList = dropIdMap[destId].list;
+    const userChoosesMoreThanAllowedNumberOfCandidates =
+      destId == "chosenCandidatesDropId" &&
+      destList.length >= pollData.maxNumRankedChoiceCount;
+    if (userChoosesMoreThanAllowedNumberOfCandidates) {
+      setPageAlert({
+        severity: "warning",
+        message: `You can only choose a max of ${pollData.maxNumRankedChoiceCount} candidates!`,
+      });
+    }
+    const isValid = !userChoosesMoreThanAllowedNumberOfCandidates;
+    return isValid;
+  };
+
+  const move = (
+    source: string[],
+    destination: string[],
+    sourceIndex: number,
+    destIndex: number
+  ): string[][] => {
+    const sourceClone = [...source];
+    const destClone = [...destination];
+    const [removed] = sourceClone.splice(sourceIndex, 1);
+    destClone.splice(destIndex, 0, removed);
+    return [sourceClone, destClone];
+  };
+
+  const boxWidth = "22.5rem";
```

Comment:
Do you guys use rems at work? I have always used `px` for things like this and height and padding etc.

Reply:
Yes, rem allows elements to scale based on root font-size. If the users font-size is larger, the box width/height should scale accordingly.

Reply:
Makes sense - I just wonder how bad it plays with other css i have written that doesn't use rem. Like some things will grow and other things wont so it might be super ugly but we can play with that later